ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -737,44 +736,86 @@ impl<'a> Assemble<'a> {
 
     fn cg_loadtraceinput(&mut self, iidx: jit_ir::InstIdx, inst: &jit_ir::LoadTraceInputInst) {
         // Find the argument register containing the pointer to the live variables struct.
-        let base_reg = ARG_GP_REGS[JITFUNC_LIVEVARS_ARGIDX];
         match self.m.type_(inst.tyidx()) {
             Ty::Integer(_) | Ty::Ptr => {
+                let avoid = RegSet::from_vec(vec![
```

Comment:
Why are we avoiding these particular registers? Should they be a `const` with a doc string at the top of the file?

Reply:
These are the caller-saved registers, which are the only registers that are safe to use for live variables before a function call (and thus the only registers referenced by stackmaps).

Yeah those could be a `const` at the top. Will fix.

Reply:
[4d4dd1f](https://github.com/ykjit/yk/pull/1318/commits/4d4dd1f925705c7da45b7a69ad4f72d61a350141)